# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/windowwatcher/nsWindowWatcher.cpp
# Commit: 3bffe760e2db
# Full Hash: 3bffe760e2db5ff009f0d798cbb63b83a68de51b
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-08-15 09:46:49
# Regressor Bug: 1562292
# File Overlap Count: 1
# Description:
#   Bug 1562292: Part 1e - Use BrowsingContext as target in IsSandboxedFrom(). r=nika
#   
#   This check always has access to an in-process DocShell which is attempting to
#   perform a load, but its target may often be cross-process, and have no
#   in-process DocShell. This patch changes the target checks to use a
# ==============================================================================

diff -r ba987e21b532 -r 3bffe760e2db toolkit/components/windowwatcher/nsWindowWatcher.cpp
--- a/toolkit/components/windowwatcher/nsWindowWatcher.cpp	Thu Aug 01 16:22:52 2019 -0700
+++ b/toolkit/components/windowwatcher/nsWindowWatcher.cpp	Thu Aug 01 16:36:32 2019 -0700
@@ -656,8 +656,8 @@
   if (parentWindow) {
     parentDocShell = parentWindow->GetDocShell();
     if (parentDocShell) {
-      nsCOMPtr<nsIDocShell> foundDocShell = do_QueryInterface(newDocShellItem);
-      if (parentDocShell->IsSandboxedFrom(foundDocShell)) {
+      if (parentDocShell->IsSandboxedFrom(
+              newDocShellItem->GetBrowsingContext())) {
         return NS_ERROR_DOM_INVALID_ACCESS_ERR;
       }
     }
